Modelica/Blocks/Math.mo

```diff
@@ -1939,7 +1939,7 @@ Otherwise the input angle <code>u</code> is wrapped to the <a href=\"https://en.
     Interfaces.IntegerOutput y "Connector of Integer output signal" annotation (
        Placement(transformation(extent={{100,-10},{120,10}})));
   equation
-    y = if (u > 0) then integer(floor(u + 0.5)) else integer(ceil(u - 0.5));
+    y = Modelica.Math.nearestInteger(u);
```

Comment:
This should be rejected since `Modelica.Math.nearestInteger(u)` is a discrete-time `Integer`.

Reply:
Even if the function had `GenerateEvents = true`, that wouldn't change the expression variability of the result, right?  Hence, isn't the only way to solve this to ensure that all inputs passed to the user-defined function are discrete-time?  For instance, one could pass both `integer(floor(u + 0.5))` and `integer(ceil(u - 0.5))` to a user-defined function which would be able to return a discrete-time `Integer`, but that function wouldn't be an abstraction worthy a place in the MSL, in my opinion.

Reply:
> This should be rejected since `Modelica.Math.nearestInteger(u)` is a discrete-time `Integer`.

I assume you mean "non-discrete time".
Note that this is true even if the function would have GenerateEvents (at least currently).

So, the proper solution seems to be to have GenerateEvents influence variability in the specification, and then it might be legal - but that's too late for MSL 4.1.0.

Reply:
Yes, I'll edit the original comment above to avoid confusion.

Reply:
I rebased and skipped that bad commit.